subsys/bluetooth/controller/CMakeLists.txt

```diff
@@ -62,7 +62,7 @@ zephyr_library_include_directories(
 
 zephyr_library_compile_options_ifdef(
   CONFIG_BT_CTLR_FAST_ENC
-  -Ofast
+  ${OPTIMIZE_FOR_SPEED_FLAG}
```

Comment:
that translates to -O2, -Ofast is -O3 AFAIK

Reply:
I know, the risk concerning this change has been elaborated on in the commit message.

Reply:
Simply put, the ble controller should be using Zephyr's "go fast" flag.

If Zephyr's "go fast" flag proves to not be "fast enough", Zephyr will need to change it's "go fast" flag, or introduce a "go even faster" flag ...

Reply:
Can we introduce the "go faster" flag instead and use it here? I am 90% sure this is not going to work for nRF51 without `-Ofast`. I can run the test though.

Reply:
well, when -Ofast was put, it was put for a reason, so you are changing behaviour here and hoping for the best. I do not think this should be merged without having verified this.

Reply:
The need to regression test has been noted in the PR description.

There is no comment about using Ofast vs. O2 so I have no way of knowing whether this was necessary or not.

Globally changing "go fast" to -Ofast is an option, and introducing "go faster" is an option, but I would prefer to first see if it is necessary.

Reply:
git blame to the rescue:

09368217d482587da55369549da5b43a6f85d156

```
commit 09368217d482587da55369549da5b43a6f85d156
Author: Vinayak Kariappa Chettimada <vich@nordicsemi.no>
Date:   Wed Sep 6 09:14:54 2017 +0200

    Bluetooth: controller: Fast enc setup with ccflags -Ofast

    To meet CPU time usage restricts inside radio ISR on nRF51
    SoCs, use ccflags -Ofast when using fast encryption setup
    implementation in the controller.

    Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>
```

Reply:
We have now tested this and the test results have shown that O2 is fast enough and that we do not need Ofast.

I would therefore like to ask that the review is re-assessed.

Reply:
@SebastianBoe would you please mind copying the results of the analysis done by @pizi-nordic in your commit message and PR description please?
I'd like this to be well documented for future reference.